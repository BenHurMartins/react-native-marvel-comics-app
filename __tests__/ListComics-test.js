/**
 * @format
 */

import 'react-native';
import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ListComics from '../src/scenes/listComics';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => jest.fn(),
    useRoute: jest.fn().mockImplementation(() => {
      return {
        params: {
          favoriteChar: {
            id: 1234,
          },
        },
      };
    }),
  };
});

describe('Redenring tests for ListComics', () => {
  it('ListComics renders correctly', () => {
    renderer.create(<ListComics />);
  });
  it('ListComics snapshot test', () => {
    const tree = renderer.create(<ListComics />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
