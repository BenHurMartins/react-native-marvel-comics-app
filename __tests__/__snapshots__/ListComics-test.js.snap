// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redenring tests for ListComics ListComics snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#EDF4ED",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
